File: backend/app/schemas/privacy.py
"""隐私授权接口模型。"""

from pydantic import BaseModel, Field

# 支持的授权类型
CONSENT_TYPES = {"camera", "album", "microphone"}


class ConsentUpdate(BaseModel):
    """更新某一项授权状态。"""

    consent_type: str = Field(..., description="授权类型：camera/album/microphone")
    granted: bool = Field(..., description="是否已授权")


class ConsentResponse(BaseModel):
    """单项授权状态响应。"""

    consent_type: str
    granted: bool

File: frontend/src/api/privacy.js
import http from './http'
import { getDeviceId } from './herb'

/** 附带设备标识的请求头。 */
function deviceHeader() {
  return { 'X-Device-Id': getDeviceId() }
}

/** 查询当前设备的隐私授权状态列表。 */
export async function listConsents() {
  const { data } = await http.get('/privacy/consents', { headers: deviceHeader() })
  return data
}

/** 更新某一项授权状态（camera/album/microphone）。 */
export async function updateConsent(consentType, granted) {
  const { data } = await http.put(
    '/privacy/consents',
    { consent_type: consentType, granted },
    { headers: deviceHeader() }
  )
  return data
}
